findspam.py

```diff
@@ -60,6 +60,14 @@
     "microsoft", "newegg", "cnet", "regex101", r"(?<!plus\.)google", "localhost", "ubuntu", "getbootstrap",
     r"jsfiddle\.net", r"codepen\.io", "pastebin", r"nltk\.org", r"xahlee\.info", r"ergoemacs\.org"
 ] + [se_dom.replace(".", r"\.") for se_dom in SE_SITES_DOMAINS])))
+URL_SHORTENER = r"(?:{})".format('|'.join(regex.escape(site) for site in (
+    '9nl.me', 'adf.ly', 'adfoc.us', 'adyou.co', 'alturl.com', 'amzn.to',
+    'bfy.tw', 'bit.do', 'bit.ly', 'bluenik.com', 'buff.ly',
+    'cl.ly', 'clkmein.com', 'dyo.gs', 'fb.me', 'goo.gl',
+    'is.gd', 'j.mp', 'ow.ly', 'post.ly', 'rurl.us', 'surl.cn.com'
+    't.co', 'tiny.cc', 'tinyurl.com', 'tr.im', 'tgig.ir',
+    'wp.me',
+    )))
```

Comment:
All of these strings need the `.` changed to `\\.`, or to be raw strings an be `\.`.

I'd suggest a variable name that explicitly states that it's a regex. So, something like `URL_SHORTENER_DOMAINS_REGEX`


While it didn't previously exist, if we're going to be detecting these domains generically, they need some type of bookending to prevent false positives. What I've been using in the blacklisted websites list is:

    \b(?<![^\W_]-)gfix\.in(?![.-][^\W_])\b

So, line 63 above could be changed to:

    URL_SHORTENER_DOMAINS_REGEX = r"(?:\b(?<![^\W_]-)(?:{})(?![.-][^\W_])\b)".format('|'.join(regex.escape(site) for site in (

If this is only going to be used in the two locations indicated in this commit, then bookending here isn't needed, because those two uses have their own, more restrictive, bookending. My concern is that if we have a variable with a general list, then it's likely to be used somewhere else sometime in the future and we'd need to remember that it needs to be bookended.

Reply:
The regex already contains the backslash; we run the whole thing through `regex.escape` (which is why the goo.gl entry has to be doctored separately afterwards).

I'll rename the variable and think about the other comments. Thanks for the feedback.

Reply:
Ahh.. sorry. I completely missed the `regex.escape()`, even though I copy & pasted it. I just didn't look for it / see it.